spec/features/dashboard_spec.rb

```diff
@@ -217,6 +219,28 @@
         expect(page).to_not have_content(other_solicitor_dr.solicitor_name)
       end
     end
+
+    scenario 'after the refresh i can still see ONLY the correct data', js: true do
+      visit defence_requests_path
+
+      within ".accepted_defence_request" do
+        expect(page).to have_content(solicitor_dr.detainee_name)
+        expect(page).to_not have_content(solicitor_dr_not_accepted.detainee_name)
+      end
+      expect(page).to_not have_selector(".created_defence_request")
```

Comment:
Agree. We’re doing this a lot and its not ideal. One of the tasks next week is to refactor/clean all this up.




Patrick Davenport		
Unboxed
Consulting
E:	patrick.davenport@unboxedconsulting.com
M:	+44 7964 292 110
T:	+44 20 7183 4264
F:	+44 20 7183 4251
17 Blossom Street
London, E1 6PL
United Kingdom
www.unboxedconsulting.com

On 24 February 2015 at 10:33:23, Andrew White (notifications@github.com) wrote:

In spec/features/dashboard_spec.rb:

> @@ -217,6 +219,28 @@
>          expect(page).to_not have_content(other_solicitor_dr.solicitor_name)
>        end
>      end
> +
> +    scenario 'after the refresh i can still see ONLY the correct data', js: true do
> +      visit defence_requests_path
> +
> +      within ".accepted_defence_request" do
> +        expect(page).to have_content(solicitor_dr.detainee_name)
> +        expect(page).to_not have_content(solicitor_dr_not_accepted.detainee_name)
> +      end
> +      expect(page).to_not have_selector(".created_defence_request")
Really don't like setting expectations on styles like this - makes tests very brittle to design changes, but I guess it's okay for the alpha

—
Reply to this email directly or view it on GitHub.